safe/gis/generic_expressions.py

```diff
@@ -207,8 +207,15 @@ def beautify_date(inasafe_time, feature, parent):
         qgis_variable.
     """
     _ = feature, parent  # NOQA
-    datetime_object = datetime.datetime.strptime(
-        inasafe_time, '%Y-%m-%dT%H:%M:%S.%f')
+    try:
+        datetime_object = datetime.datetime.strptime(
+            inasafe_time, '%Y-%m-%dT%H:%M:%S.%f')
+    except ValueError:
+        try:
+            datetime_object = datetime.datetime.strptime(
+                inasafe_time, '%Y-%m-%dT%H:%M:%S')
```

Comment:
L212 and L216, why are you not using the format described in definitions?

Reply:
I haven't used it since it's specific to that `extra_keyword`. I am planning to make a parser to string based on several formats (and put the formats in the safe/definitions/constants). But may be in the next PR, since there are some parts that have similar pattern.